File: scripts/gen_icon.mjs
#!/usr/bin/env node
/**
 * 生成 app 图标源图 (1024x1024 PNG)。
 *
 * 图标本体取自 Lucide 开源图标库（ISC 许可）：src-tauri/icons-src/lucide-key-round.svg
 * 合成方式：蓝色圆角方块背景 + 居中白色钥匙符号，再用 @resvg/resvg-js 渲染为 PNG。
 *
 * 用法：npm run icon   （生成 src-tauri/app-icon.png，随后跑 `npx tauri icon src-tauri/app-icon.png`）
 */
import { readFileSync, writeFileSync } from "node:fs";
import { dirname, join } from "node:path";
import { fileURLToPath } from "node:url";
import { Resvg } from "@resvg/resvg-js";

const root = join(dirname(fileURLToPath(import.meta.url)), "..");
const srcSvg = readFileSync(
  join(root, "src-tauri/icons-src/lucide-key-round.svg"),
  "utf8"
);

// 抽取 <svg ...> 标签内的图形元素，并把 currentColor 换成白色
const bodyMatch = srcSvg.match(/<svg[^>]*>([\s\S]*?)<\/svg>/);
if (!bodyMatch) throw new Error("无法解析 Lucide SVG");
const glyph = bodyMatch[1].replace(/currentColor/g, "#ffffff").trim();

const SIZE = 1024;
const VB = 24; // Lucide 的 viewBox
const scale = 0.625; // 符号占比（app 图标惯例 60%~70%）
const pad = (VB - VB * scale) / 2;

const svg = `<svg xmlns="http://www.w3.org/2000/svg" width="${SIZE}" height="${SIZE}" viewBox="0 0 ${VB} ${VB}">
  <rect x="0" y="0" width="${VB}" height="${VB}" rx="5.4" fill="#2563EB"/>
  <g transform="translate(${pad},${pad}) scale(${scale})" fill="none" stroke="#ffffff" stroke-width="2.6" stroke-linecap="round" stroke-linejoin="round">
    ${glyph}
  </g>
</svg>`;

const resvg = new Resvg(svg, { fitTo: { mode: "width", value: SIZE } });
const png = resvg.render().asPng();
const out = join(root, "src-tauri/app-icon.png");
writeFileSync(out, png);
console.log(`saved ${out} (${png.length} bytes)`);

File: scripts/gen_icon.py
#!/usr/bin/env python3
"""生成 app 图标源图 (1024x1024)。
蓝色圆角方块 + 白色钥匙符号（授权语义）。纯几何、无版权。
依赖：Pillow。
"""
from PIL import Image, ImageDraw

W = 1024
img = Image.new("RGBA", (W, W), (0, 0, 0, 0))
d = ImageDraw.Draw(img)

# 背景：蓝色圆角方块
d.rounded_rectangle([0, 0, W - 1, W - 1], radius=232, fill=(37, 99, 235, 255))

white = (255, 255, 255, 255)
lw = 66
cx, cy = W // 2, W // 2
hr = 132
hx = cx - 160
hy = cy

# 钥匙头：圆环
d.ellipse([hx - hr, hy - hr, hx + hr, hy + hr], outline=white, width=lw)

# 钥匙柄：向右
shaft = 330
sx = hx + int(hr * 0.55)
d.rectangle([sx, cy - lw // 2, sx + shaft, cy + lw // 2], fill=white)

# 钥匙齿：末端向下两道
bx = sx + shaft
d.rectangle([bx - 6, cy + lw // 2, bx + 34, cy + lw // 2 + 78], fill=white)
d.rectangle([bx - 56, cy + lw // 2, bx - 16, cy + lw // 2 + 54], fill=white)

img.save("src-tauri/app-icon.png")
print("saved", img.size)
